### ISSUE TITLE
FEATURE REQUEST: ssh agent should support "destination restrictions"

### ISSUE THREAD

**Issue by l4t3b0** [strongbox-password-safe/Strongbox#772]
Currently if I have more than 6 ssh keys enabled in the strongbox profile we have difficulties using them. We get "Too many authentication failures" back from the server.

OpenSSH 8.9 includes the ability to control how and where keys in ssh-agent may be used, both locally and when forwarded.

In practice it means, that if we use ssh-add with the -h option, than we can have more than 6 private keys loaded into the agent and we won't run into the trouble of the too many authentication failures.

I request to implement the same behaviour ssh-agent/ssh-add has to avoid manually disabling ssh keys in strongbox.

**Comment by strongbox-mark** [comment]
Interesting, have you seen the help article on mitigating this issue:

https://strongbox.reamaze.com/kb/ssh-agent/ssh-agent

We have a specific section on this: SSH Key Limits, Multiple Github Accounts and Key Specification

**Comment by strongbox-mark** [comment]
Interesting idea. However, since there is already a well defined solution to this issue (config), it is unlikely we can schedule work on this anytime soon.

**Comment by daletman** [comment]
I have the same issue. I create a new ssh key for every container I create on Proxmox and put the IP address into the Strongbox entry and expected Strongbox to only serve ssh keys matching the url/ip I try to ssh to.

**Comment by ZHStudio** [comment]
I completely agree with the author, this function is necessary, it is simply impossible to work with SSH keys when there are many of them, in KeePassXC there is at least a convenient manual addition of keys through a combination, but here it is a pain. I experienced high moral suffering when I used this solution. I think you should still get a refund for the software, since there are too many things that don't work the way they should and still have to switch back to KeePassXC.

**Comment by l4t3b0** [comment]
Hello Mark

any plans to extend the functionality of the strongbox's ssh agent?
I would be very happy if you would have the time to implement is
